### ISSUE TITLE
Allow `names` paramter in `Labels` to be a string or raise a more clear error message 

### ISSUE THREAD

**Issue by PicoCentauri** [lab-cosmo/equistore#110]
## Expected behavior ##

When creating a `Label` instance with only one name I would write

```python
Labels(names="species", values=np.array([(1,)]))
```

and should get

```python
Labels([(1,)], dtype=[('species', '<i4')])
```

## Actual behavior ##

I get an `ValueError` raised by `labels.py` in line 75 stating that `names parameter must have an entry for each column of the array`. The reason is that the code is treating the string a list and assumes that there have to be also 7 columns in my array.

## Solution I would like ##

Either we convert single string inputs to list (as we do it in the operations) or we raise a more clear error message stating that the input type is wrong.

Additionally, I would rephrase the error message to something like `names parameter has 7 entries but the array only has 1 column(s).`

**Comment by Luthaf** [comment]
fixed in #126
